api/src/backend/api/utils.py

```diff
@@ -147,6 +151,47 @@ def prowler_provider_connection_test(provider: Provider) -> Connection:
     )
 
 
+def prowler_integration_connection_test(integration: Integration) -> Connection:
+    """Test the connection to a Prowler integration based on the given integration type.
+
+    Args:
+        integration (Integration): The integration object containing the integration type and associated credentials.
+
+    Returns:
+        Connection: A connection object representing the result of the connection test for the specified integration.
+    """
+    if integration.integration_type in [
+        Integration.IntegrationChoices.S3,
+        Integration.IntegrationChoices.AWS_SECURITY_HUB,
+    ]:
+        try:
+            session = AwsProvider(**integration.credentials).session.current_session
```

Comment:
S3 integration allows you to pass the same credentials as the provider and has a `test_connection` method, so you don't need to setup the AWS provider.
```suggestion
            S3.test_connection(**integration.credentials)
```

I see the `test_connection` does not support passing the raw credentials and it should. I'm going to add support to that.

Reply:
Thanks, let me know when this is available